src/main/java/rx/Observable.java

```diff
@@ -5905,6 +5905,60 @@ public final void forEach(final Action1<? super T> onNext, final Action1<Throwab
      * 
      * @param keySelector
      *            a function that extracts the key for each item
+     * @param elementSelector
+     *            a function that extracts the return element for each item
+     * @param evictingMapFactory
+     *            a function that given an eviction action returns a {@link Map} instance that will be used to assign 
+     *            items to the appropriate {@code GroupedObservable}s. The {@code Map} instance must be thread-safe 
+     *            and any eviction must trigger a call to the supplied action (synchronously or asynchronously). 
+     *            This can be used to limit the size of the map by evicting keys by maximum size or access time for 
+     *            instance. If {@code evictingMapFactory} is null then no eviction strategy will be applied (and a suitable default thread-safe 
```

Comment:
I wouldn't support `null` at the surface level. Those who don't wish to have a custom eviction strategy should use the other overload.